packages/lfrqa/src/aviary/envs/lfrqa/env.py

```diff
@@ -118,68 +118,74 @@
+    @model_validator(mode="before")
+    @classmethod
+    def _validate_fields(cls, data: dict) -> dict:
+        if data.get("gold_doc_ids") and not data.get("gt_doc_ids"):
+            data["gt_doc_ids"] = data["gold_doc_ids"]
+        if isinstance(data["gt_doc_ids"], str):
+            data["gt_doc_ids"] = data["gt_doc_ids"].strip("[]").split(",")
+            data["gt_doc_ids"] = [int(_id) for _id in data["gt_doc_ids"]]
+        del data["gold_doc_ids"]
 
-        self.qid = qid
-        self.question = question
-        self.human_answer = human_answer
-        self.gt_doc_ids = gt_doc_ids
-        self.pairwise_eval_llm = pairwise_eval_llm
+        data["ideal_answer"] = data["answer"]
+        del data["answer"]
+
+        data["question_id"] = data["qid"]
+        del data["qid"]
+
+        data["options"] = []
+        data["prompt_without_options"] = True
 
-    def extract_best_answer_index(self, text: str) -> int:
+        return data
+
+    def _extract_best_answer_index(self, text: str) -> int:
         match = re.search(r"<rating>(\d+)</rating>", text)
         return int(match.group(1)) if match else 0
 
-    async def _evaluate_answer(self) -> dict:
-        """Pairwise evaluation of PaperQA vs Human answer."""
-        paper_search_ids = [int(doc.docname) for doc in self.state.docs.docs.values()]
-
-        pairwise_eval_llm = LiteLLMModel(name=self.pairwise_eval_llm)
-        pqa_answer = strip_citations(self.state.session.answer)
+    async def grade(  # type: ignore[override]
+        self,
+        proposed_answer: str,
+        paper_search_ids: list[int],
+        llm_name: str,
+        pairwise_eval_llm: LLMModel | str = CommonLLMNames.GPT_4O.value,
+    ) -> dict:
+        pqa_answer = strip_citations(proposed_answer)
         pqa_answer_index = 1 if random.random() < 0.5 else 2  # noqa: PLR2004
         data = {
             "question": self.question,
-            "answer1": pqa_answer if pqa_answer_index == 1 else self.human_answer,
-            "answer2": self.human_answer if pqa_answer_index == 1 else pqa_answer,
+            "answer1": pqa_answer if pqa_answer_index == 1 else self.ideal_answer,
+            "answer2": self.ideal_answer if pqa_answer_index == 1 else pqa_answer,
         }
 
+        if isinstance(pairwise_eval_llm, str):
+            pairwise_eval_llm = LiteLLMModel(name=pairwise_eval_llm)
+
         result = await pairwise_eval_llm.call_single(
             messages=[
                 Message(role="system", content=lfrqa_system_prompt),
                 Message(role="user", content=lfrqa_prompt_template.format(**data)),
             ],
         )
 
-        best_answer_index = self.extract_best_answer_index(result.text or "")
+        best_answer_index = self._extract_best_answer_index(result.text or "")
         if best_answer_index == pqa_answer_index:
-            winner, reward = "paperqa", self._rewards["win"]
+            winner, reward = "paperqa", 1
```

Comment:
Hardcoded the rewards... Is that ok?

Reply:
No let's not hard code, we want this to be configurable by users with different ideas of reward values